Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -418,7 +418,8 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
             let error: string = "";
             let stdout: string = "";
             let stderr: string = "";
-            const result: number = await new Promise<number>(resolve => {
+            let buildTerminated: boolean = false;
+            await new Promise<number>(resolve => {
```

Comment:
If you no longer use the number returned by `resolve(-1)` or `resolve(0)`, then this should be a `Promise<void>`.  But I don't see anything significantly different\*\* in this PR from what you're already doing.

\*\*except... (see my next comment)